pkg/adaptor/proxy/proxy.go

```diff
@@ -62,12 +66,33 @@ func NewAgentProxy(socketPath, criSocketPath string, pauseImage string) AgentPro
 		maxRetries:    defaultMaxRetries,
 		retryInterval: defaultRetryInterval,
 		pauseImage:    pauseImage,
+		tlsConfig:     tlsConfig,
 	}
 }
 
 func (p *agentProxy) dial(ctx context.Context, address string) (net.Conn, error) {
 
 	var conn net.Conn
+	var config *tls.Config
+
+	dialer := &tls.Dialer{
+		NetDialer: &net.Dialer{
+			Timeout: p.retryInterval,
+		},
+		Config: config,
+	}
+
+	if p.tlsConfig != nil && !reflect.DeepEqual(tlsutil.TLSConfig{}, *p.tlsConfig) {
+
+		logger.Printf("TLS is configured. Configure TLS dialer")
+
+		// Create a TLS configuration object
+		config, err := tlsutil.GetTLSConfigFor(p.tlsConfig)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to create tls config: %v", err)
+		}
+		dialer.Config = config
+	}
```

Comment:
Is it possible to simply this code like this?

```
	var dialer interface {
		DialContext(ctx context.Context, network, address string) (net.Conn, error)
	}

	if p.tlsConfig != nil && !reflect.DeepEqual(tlsutil.TLSConfig{}, *p.tlsConfig) {
		logger.Printf("TLS is configured. Configure TLS dialer")

		// Create a TLS configuration object
		config, err := tlsutil.GetTLSConfigFor(p.tlsConfig)
		if err != nil {
			return nil, fmt.Errorf("Failed to create tls config: %v", err)
		}
		dialer = &tls.Dialer{
			Config: config,
		}
	} else {
		dialer = &net.Dialer{}
	}
```

Reply:
Done